File: exercism/cpp/secret-handshake/secret_handshake.cpp
#include "secret_handshake.h"
#include <vector>
#include <string>
#include <bits/stdc++.h>
using namespace std;
namespace secret_handshake
{

    vector<string> commands(int code)
    {

        vector<string> signals = vector<string>();

        if ((code & 1) == 1)
        {
            signals.push_back("wink");
        }

        if ((code & 2) == 2)
        {
            signals.push_back("double blink");
        }

        if ((code & 4) == 4)
        {
            signals.push_back("close your eyes");
        }

        if ((code & 8) == 8)
        {
            signals.push_back("jump");
        }

        if ((code & 16) == 16)
        {
            reverse(signals.begin(), signals.end());
        }

        return signals;
    }

} // namespace secret_handshake

File: exercism/julia/secret-handshake/secret-handshake.jl
const handshakes = [1 => "wink", 2 => "double blink", 3 => "close your eyes", 4 => "jump"]

function secret_handshake(code)
    handshakes_result = []
    codestring = bitstring(Int8(code)) |> reverse
    for (index, action) in handshakes 
        if codestring[index] == '1'
            push!(handshakes_result, action)
        end
    end
    if codestring[5] == '1'
        handshakes_result |> reverse |> collect
    else
        handshakes_result
    end 
end